### DISCUSSION TITLE
Callback handler when scroll starts and scroll finishes.

### DISCUSSION THREAD

**Discussion by devden0702-pepper-content** [gilbarbara/react-joyride#840] [Q&A] [answered]
Hi,

I was wondering if Joyride provides any callback handler which handles when a scroll starts and when a scroll finishes when we transition from one Step to the next Step. For ex:- Our spotlight is showing in the centre of the screen and if I click next, the page scrolls to bottom and the next Step is shown in the spotlight. So, is it possible by Joyride to detect when the scroll finishes and the next spotlight is displayed.

**Comment by gilbarbara** [comment, marked as answer]
Hey @devden0702-pepper-content 

There's only one callback that is triggered when the state changes.

I haven't tried, but I think you can achieve that by tracking the step's **lifecycle** between `step:before` and `tooltip` (or `beacon` if **continuous** is false).

**Reply by devden0702-pepper-content** [reply]
I have tried the same but the lifecycle value becomes `tooltip` before the scroll transition is finished.

**Comment by gilbarbara** [comment]
Yeah, I don't remember handling the scroll to lifecycle changes, but worth the try.
What's your use case for this?

**Reply by devden0702-pepper-content** [reply]
Currently I want to set CSS property of Overflow Hidden/Unset. So as soon as the Scroll starts, I want the Overflow to be unset (i.e it will take the default scroll behaviour), and when the Scroll ends, I want the Overflow to be hidden(disable the scroll completely). The reason I want this is bcoz, the spotlight is scrolling when the user scrolls inside the Overlay. That should not be the case (my project requirement).
